pkg/detectors/typetalk/typetalk.go

```diff
@@ -0,0 +1,81 @@
+package typetalk
+
+import (
+	"context"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	//Make sure that your group is surrounded in boundry characters such as below to reduce false positives
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"typetalk"}) + `\b([0-9a-zA-Z]{64})\b`)
+	idPat  = regexp.MustCompile(detectors.PrefixRegex([]string{"typetalk"}) + `\b([0-9a-zA-Z]{32})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"typetalk"}
+}
+
+// FromData will find and optionally verify Typetalk secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+	idMatches := idPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		if len(match) != 2 {
+			continue
+		}
+		resMatch := strings.TrimSpace(match[1])
+
+		for _, idMatch := range idMatches {
+			if len(idMatch) != 2 {
+				continue
+			}
+
+			resIdMatch := strings.TrimSpace(idMatch[1])
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_Typetalk,
+				Raw:          []byte(resMatch),
+			}
+
+			if verify {
+				req, err := http.NewRequestWithContext(ctx, "POST", "https://typetalk.com/oauth2/access_token?client_id="+resIdMatch+"&client_secret="+resMatch+"&grant_type=client_credentials", nil)
```

Comment:
nit: Prefer fmt.Sprintf when there are multiple variables within the string. 
`fmt.Sprintf("https://typetalk.com/oauth2/access_token?client_id=%s&client_secret=%s&grant_type=client_credentials", resIdMatch, resMatch)`

Reply:
pushed the enhancement, thanks @ahrav!